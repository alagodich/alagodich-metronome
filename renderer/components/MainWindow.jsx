import React, {Component, PropTypes} from 'react';
import {Link} from 'react-router';
import Metronome from './Metronome.jsx';
import Voting from './Voting.jsx';
import {List as list} from 'immutable';

const propTypes = {
        // ipc: PropTypes.object,
        // children: PropTypes.object
    },
    pair = list.of('Trainspotting', '28 Days Later'),
    vote = entry => () => console.log(`I vote for ${entry}`);

class MainWindow extends Component {
    render() {
        console.log('MainWindow::render()');
        // return React.cloneElement(this.props.children, {pair: pair.toJS(), vote});
        return (
            <div>
                <Voting pair={pair} vote={vote} hasVoted={pair[0]}/>
                <Metronome />
                <Link to="/counter">to Counter</Link>
            </div>
        );
    }
}

MainWindow.propTypes = propTypes;

export default MainWindow;
