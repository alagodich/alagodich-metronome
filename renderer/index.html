<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Metronome</title>
    <!-- CSS -->
    <link href="../dist/app.css" rel="stylesheet">
  </head>

  <body>
      <div class="ui main container basic segment">
          <div class="app"></div>
      </div>
  </body>

  <script>
    require('../dist/vendors.js');
    require('../dist/app.js');
  </script>

</html>
