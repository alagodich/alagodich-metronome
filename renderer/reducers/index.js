import {combineReducers} from 'redux';
import {routerReducer as routing} from 'react-router-redux';
import counter from './counter';
import voting from './voting';

const rootReducer = combineReducers({counter, voting, routing});

export default rootReducer;
