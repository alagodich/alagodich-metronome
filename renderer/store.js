import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {hashHistory} from 'react-router';
import {routerMiddleware} from 'react-router-redux';
import rootReducer from './reducers/index.js';

const router = routerMiddleware(hashHistory),
    enhancer = applyMiddleware(thunk, router);

export default function configureStore(initialState) {
    return createStore(rootReducer, initialState, enhancer);
}
