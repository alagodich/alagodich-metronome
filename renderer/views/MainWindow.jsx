import React, {Component} from 'react';

const propTypes = {};

class MainWindow extends Component {
    render() {
        return <div>{'Main Window'}</div>;
    }
}

MainWindow.propTypes = propTypes;

export default MainWindow;
