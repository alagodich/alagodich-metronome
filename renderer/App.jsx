import MainWindow from './components/MainWindow.jsx';
import React from 'react';
import ReactDom from 'react-dom';

window.onload = () => {
    const container = document.querySelector('.app');
    if (!container) {
        return;
    }
    ReactDom.render(<MainWindow />, container);
};
