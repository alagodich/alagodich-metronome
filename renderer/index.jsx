/* eslint no-console: 0 */

import React from 'react';
import {render} from 'react-dom';
import {Provider} from 'react-redux';
import {syncHistoryWithStore} from 'react-router-redux';
import {ipcRenderer} from 'electron';
import {Router, Route, hashHistory, IndexRoute} from 'react-router';
import configureStore from './store';

// import MainWindow from './components/MainWindow.jsx';
import App from './containers/App.jsx';
import HomePage from './containers/HomePage.jsx';
import CounterPage from './containers/CounterPage.jsx';

const routes = (
    <Route path={'/'} component={App}>
        <IndexRoute component={HomePage} />
        <Route path="/counter" component={CounterPage} />
    </Route>
    ),
    store = configureStore(),
    history = syncHistoryWithStore(hashHistory, store);

ipcRenderer.on('store-update', (event, arg) => {
    console.log('store-update recieved', event, arg);
});

ipcRenderer.send('window-connected');

window.onload = () => {
    const container = document.querySelector('.app');
    if (!container) {
        return;
    }
    // render(<MainWindow ipc={ipcRenderer} />, container);
    render(
        <Provider store={store}>
            <Router history={history} routes={routes} />
        </Provider>,
        container
    );
};
