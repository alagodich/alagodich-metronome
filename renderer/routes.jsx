import React from 'react';
import {Route, IndexRoute} from 'react-router';
import App from './containers/App.jsx';
import HomePage from './containers/HomePage.jsx';
import CounterPage from './containers/CounterPage.jsx';
import {metronomePage, onInit as initMetronome} from './containers/metronomePage.jsx';
import VotingPage from './containers/VotingPage.jsx';

// export default dispatch => (
//     <Route path={'/'} component={App}>
//         <IndexRoute component={HomePage} />
//         <Route path="/counter" component={CounterPage} />
//         <Route path="/metronome" component={metronomePage()} onEnter={initMetronome(dispatch)} />
//         <Route path="/voting" component={VotingPage} />
//     </Route>
// );
export default dispatch => (
    <Route path={'/'} component={App}>
        <IndexRoute component={metronomePage()} onEnter={initMetronome(dispatch)} />
    </Route>
);
