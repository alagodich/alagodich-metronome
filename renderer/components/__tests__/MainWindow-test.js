import React from 'react';
import MainWindow from '../MainWindow';
import renderer from 'react-test-renderer';
import electron from 'electron';

jest.mock('electron', () => ({
    ipcRenderer: ({
        on: jest.fn(() => {})
    })
}));

test('MainWindow can be rendered', () => {
    const component = renderer.create(<MainWindow ipc={electron.ipcRenderer} />),
        tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
