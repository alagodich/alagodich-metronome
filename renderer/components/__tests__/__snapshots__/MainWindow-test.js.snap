exports[`test MainWindow can be rendered 1`] = `
<div>
  <div
    className="voting">
    <div
      className="voting ui buttons">
      <button
        className="ui button"
        onClick={[Function]}>
        Trainspotting
      </button>
      <button
        className="ui button"
        onClick={[Function]}>
        28 Days Later
      </button>
    </div>
  </div>
  <div
    className="metronome">
    <button
      className="ui icon button"
      onClick={[Function]}>
      <i
        className="cloud icon" />
    </button>
    <div />
  </div>
</div>
`;
